### PR TITLE
Support function_response_types in event_source_mappings

### PR DESCRIPTION
**Pull request by Schwamkrug**
Hi! This PR updates `event_source_mappings` to support partial batch responses.
https://aws.amazon.com/about-aws/whats-new/2021/11/aws-lambda-partial-batch-response-sqs-event-source/
https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html#services-sqs-batchfailurereporting

### CODE REVIEW

File: event_source_mappings.tf [hunk 1]
```diff
@@ -44,6 +44,7 @@ resource "aws_lambda_event_source_mapping" "event_source" {
   parallelization_factor             = lookup(each.value, "parallelization_factor", null)
   starting_position                  = lookup(each.value, "starting_position", length(regexall(".*:sqs:.*", lookup(each.value, "event_source_arn", null))) > 0 ? null : "TRIM_HORIZON")
   starting_position_timestamp        = lookup(each.value, "starting_position_timestamp", null)
+  function_response_types            = lookup(each.value, "function_response_types", [])
```
Comment (Schwamkrug) [inline event_source_mappings.tf:47]: I tested this out defaulting to both an empty list `[]` and to `null`. Both behaved the same. Do you have a preference?

Comment (moritzzimmer) [reply event_source_mappings.tf:47]: @Schwamkrug `null` would be a bit more consistent to the other configuration values, so I would lean to that.

Comment (Schwamkrug) [reply event_source_mappings.tf]: Sounds good. Fixed!

**Comment by moritzzimmer** [comment]
this has been released with [v6.6.0](https://github.com/moritzzimmer/terraform-aws-lambda/releases/tag/v6.6.0) - thx for the contribution @Schwamkrug!

**Pull request merged**